sigs/storage-charter.md

```diff
@@ -0,0 +1,125 @@
+# CNCF Storage SIG Charter
+
+Primary Author: Quinton Hoole
+
+2019-04-23
+
+Reviewed and contributed to by Saad Ali, Alex Chircop,
+Xing Yang, Xiang Li, and other members of the CNCF Storage Working
+Group and community.
+
+# Introduction
+
+This is the charter referred to in [CNCF
+SIGs](https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#sig-charter)
+by the CNCF TOC.
+
+# Areas Considered In Scope
+
+Storage systems and approaches suitable for and commonly used in
+[modern cloud-native
+environments](https://github.com/cncf/toc/blob/master/DEFINITION.md):
+
+*   especially where these differ significantly from storage systems
+    and approaches previously commonly used in traditional enterprise
+    data center environments,
+*   and which are not already adequately covered by other groups within the CNCF (see below).  
+*   storage includes block stores, file systems, object stores, databases, key-value stores, and related caching mechanisms. 
+
+We strive to understand the fundamental characteristics of different
+storage approaches with respect to availability, scalability,
+performance, durability, consistency, ease-of-use, cost and
+operational complexity; and relate these to their suitability to
+various cloud-native use cases. This is covered in more detail in the
+[CNCF Storage Landscape White Paper](
+http://bit.ly/cncf-storage-whitepaper).
+
+# Areas Considered Out Of Scope
+
+Anything not considered in scope above (:-)).  See also “Interfaces
+with Related Groups” below.
+
+Examples include:
+
+*   Details of general-purpose low-level storage media like magnetic
+    hard disks, solid-state storage devices, non-volatile memory, etc
+    (which tend not to be very different irrespective of whether or
+    not they’re being used in a cloud-native environment).
+*   General authentication, authorization, accounting, auditing etc
+    (even though these clearly apply to storage systems too) - because
+    AAA etc is clearly the domain of the CNCF Security SIG.
+*   Standardizing container storage interfaces - this is the domain of CSI.
+*   Defining storage abstraction APIs for container orchestrators -
+    these are the domains of the Storage SIGs specific to each
+    orchestrator, e.g. Kubernetes Storage SIG.
+
+# SIG Mission Statement
+
+To enable widespread and successful storage of persistent state in
+cloud-native environments through:
+
+1. Providing valuable and objective information to the TOC, End Users
+   and Projects of the CNCF regarding areas considered in scope (see
+   above).
+2. Collaborating effectively with other related groups (see below).
+3. Helping to maintain the continued health of CNCF Storage Projects
+   (see below)
+4. Identifying and filling gaps in the landscape of CNCF Storage
+   Projects.
+
+# Current CNCF Storage Projects
+
```

Comment:
Do we expect this list of projects to be updated over time, or is this a snapshot? I am worrying about the maintainability of this list, plus also the list [here](https://github.com/cncf/toc/blob/master/sigs/cncf-sigs.md#initial-sigs)

Reply:
Agreed. We need to decide where the canonical mapping from CNCF projects to
CNCF SIGs lives. I'll leave it here for now, but we can move it to a
consolidated location once that has been decided.